packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol

```diff
@@ -619,6 +620,121 @@ contract L1CrossDomainMessenger_Test is Bridge_Initializer {
         assertTrue(l1CrossDomainMessenger.paused());
         assertEq(l1CrossDomainMessenger.paused(), superchainConfig.paused());
     }
+
+    /// @dev Tests that sendMessage succeeds with a custom gas token when the call value is zero.
+    function test_sendMessage_customGasToken_noValue_succeeds() external {
+        // Mock the gasPayingToken function to return a custom gas token
+        vm.mockCall(
+            address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(2))
```

Comment:
nit, mocked gas token should comply to spec
```suggestion
            address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(18))
```

Reply:
Looks like you got most of them, seems 3 are left when I search for `abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(2))`